spec/models/user_spec.rb

```diff
@@ -0,0 +1,7 @@
+require 'spec_helper'
+
+describe User do 
+  it { should validate_presence_of(:username)}
+  it { should validate_uniqueness_of(:username)}
```

Comment:
I suggest making the spacing consistent here.

`it { should validate_uniqueness_of(:username) }`